src/main/java/ch/njol/skript/structures/StructEvent.java

```diff
@@ -0,0 +1,115 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.structures;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.NoDoc;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.parser.ParserInstance;
+import org.bukkit.event.Event;
+import org.bukkit.event.EventPriority;
+import org.eclipse.jdt.annotation.Nullable;
+import org.skriptlang.skript.lang.entry.EntryContainer;
+import org.skriptlang.skript.lang.structure.Structure;
+
+@NoDoc
+public class StructEvent extends Structure {
+
+	static {
+		Skript.registerStructure(StructEvent.class,
+				"[on] <.+> [with priority (1:lowest|2:low|3:normal|4:high|5:highest|6:monitor)]");
```

Comment:
Not in the scope of this PR, but do you happen to know why `with priority lowest` was chosen over `with lowest priority`?

Reply:
My guess would be to a) make the pattern of all priorities consistent. b) the previous strategy of picking out the priority was hardcoded, unflexable and basic. But I don't know for sure